entities/pool.go

```diff
@@ -331,32 +354,35 @@ func (p *Pool) swap(zeroForOne bool, amountSpecified *utils.Int256, sqrtPriceLim
 			step.tickNext = utils.MaxTick
 		}
 
-		step.sqrtPriceNextX96, err = utils.GetSqrtRatioAtTickV2(step.tickNext)
+		err = utils.GetSqrtRatioAtTickV2(step.tickNext, &step.sqrtPriceNextX96)
 		if err != nil {
 			return nil, err
 		}
-		var targetValue *utils.Uint160
+		var targetValue utils.Uint160
 		if zeroForOne {
 			if step.sqrtPriceNextX96.Cmp(sqrtPriceLimitX96) < 0 {
-				targetValue = sqrtPriceLimitX96
+				targetValue.Set(sqrtPriceLimitX96)
 			} else {
-				targetValue = step.sqrtPriceNextX96
+				targetValue.Set(&step.sqrtPriceNextX96)
 			}
 		} else {
 			if step.sqrtPriceNextX96.Cmp(sqrtPriceLimitX96) > 0 {
-				targetValue = sqrtPriceLimitX96
+				targetValue.Set(sqrtPriceLimitX96)
 			} else {
-				targetValue = step.sqrtPriceNextX96
+				targetValue.Set(&step.sqrtPriceNextX96)
 			}
 		}
 
-		state.sqrtPriceX96, step.amountIn, step.amountOut, step.feeAmount, err = utils.ComputeSwapStep(state.sqrtPriceX96, targetValue, state.liquidity, state.amountSpecifiedRemaining, p.Fee)
+		var nxtSqrtPriceX96 utils.Uint160
```

Comment:
nxt typo?

Reply:
um I don't get your question, it's for the price in next loop so I named it like that